packages/components/src/components/navigation-item/navigation-item.ts

```diff
@@ -0,0 +1,297 @@
+  @property({ reflect: true }) size: 'base' | 'larger' | 'smaller' = 'base';
+
+  /** The navigation item's orientation. If true, properties below this point are not used. */
+  @property({ type: Boolean, reflect: true }) horizontal = false;
+
+  /** *Vertical Only: Appends a chevron to the right side of a navigation item. */
+  @property({ type: Boolean, reflect: true }) chevron = false;
+
+  /** *Vertical Only: Adds additional padding to navigation item's left side. */
+  @property({ type: Boolean, reflect: true }) indented = false;
+
+  /** *Vertical Only: Adds additional padding to navigation item's left and right sides. */
+  @property({ type: Boolean, reflect: true }) relaxed = false;
+
+  /** *Vertical Only: Adds additional padding to navigation item's left and right sides. */
+  @property({ type: Boolean, reflect: true }) divider = false;
+
+  /** *Vertical Only if 'children' slot and no 'href': Reflects HTML details element state and allows control from parent. */
+  @property({ type: Boolean, reflect: true }) open = false;
+
+  private handleMouseEnter() {
+    this.emit('sd-mouse-enter');
+  }
+
+  private handleMouseLeave() {
+    this.emit('sd-mouse-leave');
+  }
+
+  private handleClickSummary(event: MouseEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (!this.disabled) {
+      this.open = !this.open;
+      this.emit('sd-click', { detail: { open: this.open } });
+    }
+  }
+
+  private handleClickButton(event: MouseEvent) {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+    } else {
+      this.emit('sd-click');
+    }
+  }
+
+  private isButton(): boolean {
+    return !this.href && !this.hasSlotController.test('children');
+  }
+
+  private isLink(): boolean {
+    return !!this.href;
+  }
+
+  private isAccordion(): boolean {
+    return !this.href && this.hasSlotController.test('children');
+  }
+
+  private calculatePaddingX(): string {
+    if (this.horizontal) return '';
+    if (this.relaxed && this.indented) return 'pl-8 pr-4';
+    if (this.relaxed) return 'px-4';
+    if (this.indented) return 'pl-4';
+    return '';
+  }
+
+  render() {
+    const slots = {
+      label: this.hasSlotController.test('[default]'),
+      'icon-left': this.hasSlotController.test('icon-left'),
+      'icon-right': this.hasSlotController.test('icon-right'),
+      main: this.hasSlotController.test('main'),
+      description: this.hasSlotController.test('description'),
+      children: this.hasSlotController.test('children')
+    };
+    const tag = this.isLink() ? literal`a` : slots['children'] ? literal`summary` : literal`button`;
+
+    // styles
+    const horizontalPaddingBottom = this.horizontal ? 'pb-2' : 'pb-3';
+    const hostDisplayStyle = this.horizontal
+      ? html`<style>
+          :host {
+            display: inline-block;
+          }
+        </style>`
+      : html`<style>
+          :host {
+            display: block;
+          }
```

Comment:
I wouldn't mess around here. Bringing them in a vertical column or horizontal row is the implementer's job and would remove that. Most of the time they will use `flex` all the way.

Reply:
![image](https://github.com/solid-design-system/solid/assets/39494579/a61edad2-c007-44fc-a384-bae8799d5447)


For a horizontal item, is it desired that the item simply grows to the width of its parent?